tests/components/climate/test_init.py

```diff
@@ -953,6 +955,65 @@ async def test_no_issue_no_aux_property(
     ) not in caplog.text
 
 
+async def test_humidity_validation(
+    hass: HomeAssistant, register_test_integration: MockConfigEntry
+) -> None:
+    """Test validation for humidity."""
+
+    class MockClimateEntityTemp(MockClimateEntity):
+        """Mock climate class with mocked aux heater."""
+
+        _attr_supported_features = (
+            ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.PRESET_MODE
+            | ClimateEntityFeature.SWING_MODE
+            | ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            | ClimateEntityFeature.TARGET_HUMIDITY
+        )
+        _attr_target_humidity = 50
+        _attr_target_temperature = 15
+        _attr_target_temperature_high = 18
+        _attr_target_temperature_low = 10
+        _attr_target_temperature_step = PRECISION_WHOLE
+
+        def set_humidity(self, humidity: int) -> None:
```

Comment:
What is this method doing? Seems like the test runs the same without this

Reply:
It would set the humidity if the service call would not have failed. I put it in for clarity how the entity would look like, not that it's going to use it.